src/test/java/persistence/entity/DefaultEntityManagerTest.java

```diff
@@ -74,15 +75,54 @@ void persist() {
     }
 
     @Test
-    @DisplayName("엔티티를 삭제한다.")
-    void remove() {
+    @DisplayName("엔티티를 영속성 컨텍스트에 등록하고 flush() 한다.")
+    void persistAndFlush() {
+        // given
+        final EntityManager entityManager = DefaultEntityManager.of(jdbcTemplate);
+        final EntityWithOnlyId entity = new EntityWithOnlyId(1L, "Jaden", 30, "test@email.com", 1);
+
+        // when
+        entityManager.persist(entity);
+        entityManager.flush();
```

Comment:
```suggestion
        entityManager.persist(entity);
        entityManager.remove(entity);
        entityManager.flush();
```

정수님은 위 과정을 거칠때 어떻게 작동되기를 기대하시나요?

Reply:
정수님이 정의하시는 SAVING의 단계는 무엇인가요?

Reply:
id 를 db에서 생성하는 경우, persist() 호출하고 db에 아직 반영되지 않은 상태가 SAVING 상태이고, db에 영속화가 진행된 상태를 MANAGED로 봤습니다.
두 상태 다 MANAGED 로 보고 있어서 해당 부분 수정하였습니다.

Reply:
id를 db에서 생성하지 않는 테스트와 flush()를 실행해야 db에 반영되고 id까지 생성된 것을 비교하기 위해서 추가한 테스트입니다.
persist() -> flush() 를 실행해서 db에 데이터가 생성된 것을 기대했습니다.

Reply:
```
    @Id
    private Long id;
```
엔티티가 DB생성전략을 사용하지 않을때, 
```
entityManager.persist(entity);
entityManager.remove(entity);
entityManager.flush();
```
해당 순서대로 로직이 진행될때, `엔티티가 제거 가능한 상태가 아닙니다.` 에러가 나올것 같아요.
이부분에 대해서 이야기 해보고 싶었습니다 ㅎ

---

`SAVING`은 DB와 동기화 되고 있는 사이의 단계라고 저는 생각하고 있는데요, 
persist가 되고 나면 `MANAGED`상태이어야 하지 않나 생각하고 있었어요. ㅎ

Reply:
> db에 영속화가 진행된 상태를 MANAGED로 봤습니다.

`@ID`생성 전략 DB 사이드가 아니라 어플리케이션 사이드에 있더라도, 저는 persist하고 나면 `MANAGED`되고 있는것이라고 생각하고 있어요. 어플리케이션에서 `id`에 대한 고유한 값을 잘 넣었을 거라는 기대를 하는게 자연스럽지 않나 생각하기 때문인데요. 
만약, 지연 insert로 인한 `id`가 중복이라면 그에 대한 **책임은 어플리케이션**에 있기때문에 DB exception을 받는게 당연하다고 생각했어요. 그렇게 되어야 롤백에 대한 전략도 구현할수 있을것 같구요. 

이러한 관점에 대해서 정수님 생각은 어떠신가요?

Reply:
<img width="915" alt="스크린샷 2024-10-26 오후 7 48 46" src="https://github.com/user-attachments/assets/a856b4d6-a79a-4a40-8fb7-c0b540ec1a58">

이 부분을 보고 자의적으로 SAVING 상태는 쿼리가 실행되기 전 상태라고 막연히 생각했던 것 같아요.
근데 태화님 코멘트를 보고 생각해보니까 LOADING 중 상태가 조회 쿼리가 실행중인 상태인것처럼 쿼리 실행 시간이 오래 걸릴때(?) 그 순간의 단계가 아닐까 하는 생각이 드네요. 그렇다면 psersist() 하고 바로 remove() 해도 문제 없을 것 같네요.